### PR TITLE
[BF16] Terminal Softmax in FP32 precision to maintain accuracy

### PR DESCRIPTION
**Pull request by alexey-varyzgin**
### Details:
 - *An accuracy issue is in SoftMax output layer, and returns to nominal value if perform the SoftMax in FP32 mode. The accuracy issue is for INT8 + BF16 mtcnn-p model.*
 - *...*

### Tickets:
 - *50338*

### CODE REVIEW

**Comment by alexey-varyzgin** [comment]
@dmitry-gorokhov and @maxnick could you review this PR

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp [hunk 1]
```diff
@@ -1200,6 +1200,11 @@ void MKLDNNGraph::EnforceBF16() {
     if (implication(isQuantized(), config.manualEnforceBF16)) {
         for (auto &node : graphNodes) {
             if (node->getType() != Input && node->getType() != Output) {
+                //  Softmax terminal layer will be executed in FP32 precision to maintain accuracy
+                if (node->getType() == Softmax &&
+                    node->getChildEdgeAt(0)->getChild()->getType() == Output) {
+                    continue;
```
Comment (maxnick) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:1206]: What if we do not have an output node on port 0, but there is one on the first port? Should we check all edges?

Comment (alexey-varyzgin) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp]: Done

**Review by maxnick — APPROVED** [review]

**Comment by dmitry-gorokhov** [comment]
In general I ok with the impl as a solution to maintain accuracy, Major comment: we have to cover such scenarios by tests (which will analyze exec graph and check that SoftMax was executed in FP32 precision) in order not to break this behavior in future.

**Comment by alexey-varyzgin** [comment]
Refactored in https://github.com/openvinotoolkit/openvino/pull/6114

**Pull request closed without merging**